### PR TITLE
Add support for Symbol.toStringTag (closes #21).

### LINKED ISSUE #21: Properly implement toStringTag
[W3C Bug Tracker](https://www.w3.org/Bugs/Public/show_bug.cgi?id=28244) has a discussion going on how to implement it. It's also not supported in a stable node yet, need for v6 for it anyways.

### PR DESCRIPTION
**Pull request by droooney**

### CODE REVIEW

**Comment by TimothyGu** [comment]
This isn't actually compliant to the Web IDL spec, which says the prototype's \@\@toStringTag must be the interface's name + `'Proto'`. But from @domenic in https://github.com/nodejs/node/pull/10906#issuecomment-273957809:

> Regarding toStringTag, this is actually a controversial part of the Web IDL spec. See https://www.w3.org/Bugs/Public/show_bug.cgi?id=28244. I would suggest following Chrome's version, of just putting it as a data property on the prototype, but I am somewhat biased there :).

Regardless, you probably should remove the existing `prototype.toString` function.

**Comment by domenic** [comment]
Yes, I am much happier with this than the version in the Web IDL spec, which nobody implements. But yes, we need to remove the toString function that already exists.

**Comment by droooney** [comment]
Should I set a getter according to the specs (add `'Prototype'` for the prototype) then (which seems controversial to me as well)?

**Comment by domenic** [comment]
No, as I said, I am much happier with this version than the version in the Web IDL specs. The only change is that we need to remove the toString function that we're currently installing on every class.

**Comment by droooney** [comment]
No problem :) I personally like this version more as well.

**Comment by Sebmaster** [comment]
We need to keep toString for the stringifier and impl toString method. Only the stringtag part should be removed.

**Comment by droooney** [comment]
Oops, sorry... Ridiculously long commit history for such a small change :)

File: lib/constructs/interface.js [hunk 1]
```diff
@@ -365,18 +365,15 @@ Interface.prototype.generateToString = function () {
     }
   }
   if (!hasToString) {
```
Comment (domenic) [inline lib/constructs/interface.js]: This condition is now flipped; it should be `if (hasToString)`.

I'll fix this before merging.

**Comment by domenic** [comment]
I ended up doing this a bit differently; it turns out that whole generateToString() function is no longer needed. See https://github.com/jsdom/webidl2js/commit/0590e5a8b3e91edf39d40209795ff41b87787bfe. Thanks for the kick-start though!

**Comment by droooney** [comment]
No problem :) Thanks for awesome jsdom!

**Pull request closed without merging**